%% 
%% Copyright 2007-2020 Elsevier Ltd
%% 
%% This file is part of the 'Elsarticle Bundle'.
%% ---------------------------------------------
%% 
%% It may be distributed under the conditions of the LaTeX Project Public
%% License, either version 1.2 of this license or (at your option) any
%% later version.  The latest version of this license is in
%%    http://www.latex-project.org/lppl.txt
%% and version 1.2 or later is part of all distributions of LaTeX
%% version 1999/12/01 or later.
%% 
%% The list of all files belonging to the 'Elsarticle Bundle' is
%% given in the file `manifest.txt'.
%% 
%% Template article for Elsevier's document class `elsarticle'
%% with harvard style bibliographic references

\documentclass[final,times,twocolumn,article]{elsarticle}

%% For including figures, graphicx.sty has been loaded in
%% elsarticle.cls. If you prefer to use the old commands
%% please give \usepackage{epsfig}

%% The amssymb package provides various useful mathematical symbols
\usepackage{amssymb}
\usepackage{lipsum}

%% The amsthm package provides extended theorem environments
%% \usepackage{amsthm}

%% The lineno packages adds line numbers. Start line numbering with
%% \begin{linenumbers}, end it with \end{linenumbers}. Or switch it on
%% for the whole article with \linenumbers.
%% \usepackage{lineno}

%% You might want to define your own abbreviated commands for common used terms, e.g.:
\newcommand{\kms}{km\,s$^{-1}$}
\newcommand{\msun}{$M_\odot}

\journal{Astronomy $\&$ Computing}


\begin{document}

\begin{frontmatter}

%% Title, authors and addresses

%% use the tnoteref command within \title for footnotes;
%% use the tnotetext command for theassociated footnote;
%% use the fnref command within \author or \affiliation for footnotes;
%% use the fntext command for theassociated footnote;
%% use the corref command within \author for corresponding author footnotes;
%% use the cortext command for theassociated footnote;
%% use the ead command for the email address,
%% and the form \ead[url] for the home page:
%% \title{Title\tnoteref{label1}}
%% \tnotetext[label1]{}
%% \author{Name\corref{cor1}\fnref{label2}}
%% \ead{email address}
%% \ead[url]{home page}
%% \fntext[label2]{}
%% \cortext[cor1]{}
%% \affiliation{organization={},
%%            addressline={}, 
%%            city={},
%%            postcode={}, 
%%            state={},
%%            country={}}
%% \fntext[label3]{}

\title{Cerca d'inhibidors amb ML}

%% use optional labels to link authors explicitly to addresses:
%% \author[label1,label2]{}
%% \affiliation[label1]{organization={},
%%             addressline={},
%%             city={},
%%             postcode={},
%%             state={},
%%             country={}}
%%
%% \affiliation[label2]{organization={},
%%             addressline={},
%%             city={},
%%             postcode={},
%%             state={},
%%             country={}}

\author[first]{Carla Feliu}
\affiliation[first]{organization={University of Vic - Central University of Catalonia},%Department and Organization
            addressline={}, 
            city={Vic},
            postcode={}, 
            state={},
            country={}}

\begin{abstract}
%% Text of abstract
Example abstract for the astronomy and computing journal. Here you provide a brief summary of the research and the results. RESUM DE TOT
\end{abstract}

%%Graphical abstract
%\begin{graphicalabstract}
%\includegraphics{grabs}
%\end{graphicalabstract}

%%Research highlights
%\begin{highlights}
%\item Research highlight 1
%\item Research highlight 2
%\end{highlights}

\begin{keyword}
%% keywords here, in the form: keyword \sep keyword, up to a maximum of 6 keywords
keyword 1 \sep keyword 2 \sep keyword 3 \sep keyword 4

%% PACS codes here, in the form: \PACS code \sep code

%% MSC codes here, in the form: \MSC code \sep code
%% or \MSC[2008] code \sep code (2000 is the default)

\end{keyword}


\end{frontmatter}

%\tableofcontents

%% \linenumbers

%% main text

\section{Introduction}
\label{introduction}

Proteases, also known as peptid hydrolases, are enzymes capable of catalysing hydrolytic reactions that degrade protein molecules to peptides, and finally to free amino acids. They also carry out proteolytic reactions, and regulate various enzymatic cascades that form part of metabolic cycles. \cite{Ramos2019} The role they play is key in carrying out vital biological processes, such as the regulation of various cellular processes as well as differentiation, gene expression and cell death. Protests are a very broad group of enzymes that are differentiated by, among other things, the type of amino acid residue in their catalytic site. \cite{Ramos2019}

Cystein proteases (also known as thiol proteases), are one of the types of proteases that exist, and are characterised by catalysing the breakdown of proteins by cleavage of peptide bonds using a cysteine nucleophile tilo. These proteases are classified into two types according to their location within the organism; Catepsins, located in the lysosome, and Calpains, located in the cytosol. \cite{Gupta2020}

Although these proteases have a key role in vital processes, when there is overexposure or dysregulation in pathological conditions, they can contribute to the development of many diseases.  In several studies, it has been shown that cathepsins, located in the lysosome, are related to tumour progression. \cite{Berdowska2004} \cite{Mohamed2006}. These cystein proteases allow cancer cells to attack nearby tissues, blood and lymphatic vessels and metastasise to peripheral tissues. \cite{Gupta2020} \cite{Gocheva2006} For this reason, the search for inhibitors of cystein proteases has become an objective in medical and pharmacological research, in order to develop new therapeutic opportunities. \cite{cath}

The interest of cystein proteases as targets is related to the fact that by working with their inhibition, it is possible to find therapies for many of the diseases they are involved in. They are also interesting targets due to the reactive cisterna active site, as the high nucleophilicity of the cisterna thiol, under physiological conditions, provides an ideal anchoring site for small electrophilic molecules. \cite{Maurais2019} 

The inhibition process is defined by the effectiveness of the binding of a protein to the selected targets. This depends on the reaction equilibrium between the functional groups of the drug and the active site residues of the protein. This balance is determined by the type of interaction that is created, and can be either covalent or non-covalent interactions. 
Covalent bonds are usually formed by the interaction of certain amino acids, including the nucleophilic cisterna and a reactive functional group of the ligand. They are more effective due to their long permanence in the active site, high binding affinity and high selectivity and specificity, resulting in complete inhibition of the target compared to the reversible effect of non-covalent drugs. \cite{Aljoundi2020} 

The drug development process is a very elaborate process, which can involve a lot of time and resources. It begins with the search for and characterisation of a possible biological target for a specific disease or enzyme, and then the process of creating the most suitable therapeutic compound begins. Before reaching drug development, many molecular properties must be optimised to identify and validate the target, as this is one of the most important steps in drug development. \cite{Hughes2011}
Historically, this search was guided by the calculations and intuition of expert scientists, but with the great advances in new technologies and the increase in resources, it has been possible to apply new methods that are much faster and more reliable. \cite{Hop2018}

One of the most used methods currently for this drug development, and which has allowed a great advance in this field of medical research, making the process less expensive and more efficient, is machine learning (ML). ML is a branch of artificial intelligence (AI) that applies computer algorithms with the ability to learn themselves from dirty, raw data, to then perform a specific task. Within ML methods, the most widely used and expanding is deep learning (DL). It is inspired by the information processing patterns of the human brain and is designed using multiple layers of algorithms (artificial neural networks, ANNs), each of which makes an interpretation of the data it has received. \cite{Alzubaidi2021} 

Artificial neural networks, inspired by the human brain, are created to simulate a network of model neurons on a computer. By applying algorithms that mimic the processes of real neurons, models are generated that learn to solve many types of problems. The networks are composed of artificial neuron units, capable of transmitting signals to each other.(what are artificial networks ). Each reception of a signal in the neurons is associated with a weight that varies as learning progresses, which can increase or decrease the strength of the signal that will then be sent to the other neurons. These signals pass the multilayers of artificial neurons until they reach the final layer, where the result of these connections is the prediction of the algorithm. \cite{Talevi2020}

The chemprop Architecture is a neural network built for the prediction of molecular properties that includes a directed message passing neural network (D-MPNN) module for molecular feature extraction and a feed-forward neural network (FNN) for property prediction. This model takes molecular SMILES as input and extracts all the atomic and bonding characteristics of the molecules to generate a single vector of characteristics which, by entering it into the FNN, can predict the activity of the candidate molecules. \cite{Wang2022}

Given the growing interest in cysteine proteases as targets for therapeutic treatments and the availabikity of new machine learning tools for these drug discovery processes, three questions are raised in this work.

\begin{itemize}
    \item Is it possible to identify covalent targets that can serve as inhibitors for cysteine proteases using ANN?
    \item What is the best way to train a model using ANN?
    \item Is it possible to identify molecular patterns or specific molecules that can help us in finding cysteine protease inhibitors in two completely different databases?
    
\end{itemize}

%%\section{Objectives}

We will use a ML model, ANN, applying a known dataset of compounds that interact with cysteine proteases. This dataset will be used to train our model, allowing us to make predictions from two different databases, that lead us to solve the proposed questions.

Specific objectives:

\begin{enumerate}
    \item Identify a database with known targets of cysteine proteases that includes inhibitory activity, which will serve as the training data of our ML model. 
    \item Download the database and extract the (Simplified Molecular Input Line Entry Sistem)SMILES information. Prepare the required dataframe structure for working with the ML algorithm, obtaining various features for each molecule and its inhibitory activity (IC50).
    \item Train the chemprop model using various approaches, evaluating which one makes the model more stable for our predictions. 
    \item Identify suitable databases to obtain the molecules for our predictions. Similar to the training dataframe, construct the structure necessary to make the predictions with our training model. 
    \item Analyze the data generated from the predictions made by the ML model, checking the correlation of the data and their similarity to understand if the predictions obtained are suitable for our questions raised. 
    \item Detect patterns of molecules that could be optimal for working in the discovery of cystein protease inhibitors. 
\end{enumerate}

\section{Methods}

\subsection{Datasets}

Per realitzar aquest estudi, hem treballat en tres bases de dades diferents per extreure la informació necessària per realitzar el nostre estudi. L’extracció de les dades s’ha fet depenent de cada base de dades i es detallara mes endavant.  

\subsubsection{ChEMBL}
Chembl es una base de dades de petites molècules bioactives similars a fàrmacs, que contenen propietats calculades i bioactivitats. Les dades són extretes de literatura científica i participen en el descobriment de fàrmacs. \cite{chemblweb}

El nostre interès en aquesta base de dades es tracta d’obtenir informació de molècules que puguin interaccionar amb cystein proteases per tal d’obtenir un punt de referència sobre l'activitat d'interacció que ens permeti entrenar el nostre model de ML. 
Per accedir a la informació que ens interessa per el nostre estudi, hem filtrat dins la base de dades de ChEMBL amb el terme “cystein proteasa” i hem extret els 129 target ID   resultants. Un cop obtinguts els targets de cystein proteasa, amb el mòdul requests de python hem obtingut informació de totes les molècules que tenen interacció amb cada un dels lligands, junt amb la seva activitat d’interacció, el valor IC50, que hem utilitzat per poder entrenar el model ML. 

\subsubsection{ChemDiv}

ChemDiv és una organització d’investigació reconeguda mundialment en solucions de descobriment de fàrmacs. Tenen identificades diverses llibreries específiques dirigides a abordar diverses dianes, dominis proteics, processos cel·lulars... \cite{chemdivweb}

D’aquesta base de dades hem obtingut la llibreria “Cysteine Targeted Covalent Library”, amb 39,301 compostos que tenen warheads específics per reaccionar amb cisteïna. L’objectiu d’utilitzar aquesta llibreria és, aplicant el nostre model entrenat, trobar possibles compostos que tinguin una activitat d’interacció significativa amb la cisteïna per trobar un patró que ens pugui servir per entendre quin tipu de molecula seria un bon inhibidor. 

\subsubsection{Zinc}

ZINC

\subsection{Data preparation and curation}

Per dur a terme el procés d’entrenament de models de Machine learning de manera efectiva, és molt important tenir un conjunt de dades adequat i degudament preparat. Per tant, es requereix d’una preparació prèvia. 
Aquesta estructura de dades ha de tenir una primera columna que contingui els SMILES de les molècules, features que descriguin les molècules, i finalment, en una ultima fila l’activitat d’interacció d’aquestes molècules, el valor IC50. La preparació de dades inclou també la realització d’un anàlisis exhaustiu de l’estructura de les dades, la qual cosa resulta fonamental per entendre a fons les dades, i així dur a terme un anàlisis de resultats més precís.

\subsubsection{Train}



\section{Results}
%%\label{Results}

RESULTATS DE LA CARLA 

\begin{table}
\centering
\begin{tabular}{l c c c} 
     \hline
     Train & Initial dim. & Data curated & Norm.\\ 
     \hline
     Train 1 & 19,262 & 19,172 & No\\
     \hline
     Train 2 & 19,262 & 11,703 & No \\
     \hline
     Train 3 & 19,262 & 11,703 & Z-Score\\ 
     \hline
     Train 4 & 19,262 & 11,703 & MinMaxScale\\ 
     \hline
    \end{tabular}
\caption{Train results
}
\label{Table1}
\end{table}


\subsection{Subsection title}


%%\begin{figure}
 %%   \centering 
%%	\includegraphics[width=0.4\textwidth, angle=-90]{ASCOM_journal_cover.pdf}	
%%	\caption{Astronomy \& Computing journal cover} 
%%	\label{fig_mom0}%
%%\end{figure}



\section{Discussion}
%%\label{}


\section{Conclusions}


\section*{Acknowledgements}
Thanks to ...

%% The Appendices part is started with the command \appendix;
%% appendix sections are then done as normal sections
\appendix

\section{Appendix title 1}
%% \label{}

\section{Appendix title 2}
%% \label{}

%% If you have bibdatabase file and want bibtex to generate the
%% bibitems, please use
%%

\section{Bibliography}
\bibliographystyle{unsrtnat} 
\bibliography{library}



\end{document}

\endinput
%%
%% End of file `elsarticle-template-harv.tex'.
